architecture/evm/error_normalizer.go

```diff
@@ -56,7 +56,31 @@ func ExtractJsonRpcError(r *http.Response, nr *common.NormalizedResponse, jr *co
 			msg += " Data: " + fmt.Sprintf("%v", err.Data)
 		}
 
-		// Infer from known status codes
+		//----------------------------------------------------------------
+		// 1) "Unsupported" errors
+		//----------------------------------------------------------------
+
+		if r.StatusCode == 415 || code == common.JsonRpcErrorUnsupportedException || // By HTTP status code or explicit JSON-RPC error code
+			code == -32004 || code == -32001 || // direct codes from upstream
+			strings.Contains(msg, "Unsupported method") ||
+			strings.Contains(msg, "not supported") ||
+			strings.Contains(msg, "method is not whitelisted") ||
+			strings.Contains(msg, "is not included in your current plan") {
+			return common.NewErrEndpointUnsupported(
+				common.NewErrJsonRpcExceptionInternal(
+					int(code),
+					common.JsonRpcErrorUnsupportedException,
+					err.Message,
+					nil,
+					details,
+				),
+			)
+		}
```

Comment:
In recent commit had to move this to avoid premature detection when message is only "not found" e.g. from Tenderly, we had to move this in different location.

https://github.com/erpc/erpc/pull/204/commits/6df5524559940926102bf3f62ffed6c2e6dae2ce

Reply:
gotcha. I moved the `unsupported` error normalization logic below `not found` block and also put a comment for why we want to have it there.